### DISCUSSION TITLE
Possible to hook Bevy into already existing window?

### DISCUSSION THREAD

**Discussion by ericgroom** [bevyengine/bevy#4838] [Q&A] [answered]
Bevy/Windows development noob here. I'm interested in creating something akin to Wallpaper Engine and would love to use Bevy as a renderer. I already have a handle to a window using `windows-rs` and [voodoo](https://www.codeproject.com/articles/856020/draw-behind-desktop-icons-in-windows-plus) and am wondering if it's easy to pass this to Bevy. If not I would appreciate some guidance towards a better library lest I use raw Gdi calls... This is pretty much just a personal project so supporting other platforms is not necessary.

**Comment by bjorn3** [comment, marked as answer]
This should be possible. I don't think it is possible to keep using winit for window handling if create the window handle yourself. Bevy is modular though, so you can disable bevy_winit and write your own window handling code. I'm not quite sure how to make bevy_render detect your window. You can look at the place bevy_winit is used within bevy_render and do something similar I think.

**Reply by ericgroom** [reply]
Gotcha, that is kind of in line with what I found when I briefly look through the code. So I would need to write a new plugin. Shouldn't be too bad to replace winit as it seems like it's a pretty small module, hopefully I do not need to also fork `bevy-renderer`. If that works I should be able to just publish a crate with a plug-in for rendering to the wallpaper layer which would be cool. I'll update this thread with my findings. 

I got this working. Basically I forked bevy-winit into my own project, and customized `create_window`. I wasn't ever able to get it so that bevy rendered directly to a `HWND`, but I was able to use winit to then create a child window under that `HWND` which I had no problems rendering to. I'm not going to open source for now as it's super messy but works for my use case; I stripped a bunch of stuff out of bevy-winit for my use case but it's not really relevant to it working.

**Reply by mike-mortimer** [reply]
Looking into this myself - did you ever publish your solution with the customized `create_window` ?  I'm trying to work out how to attach Bevy as a child Win32 window, in a parent Win32 window, alongside other UI content.

**Reply by ericgroom** [reply]
@mike-mortimer I hadn't but just for you I made https://github.com/ericgroom/desktop public! `wallpaper_render_plugin` is the fork of `bevy_winit`. The most important file is `winit_windows.rs` where I edited `create_window` to create a child window. Hope this helps!

**Reply by mike-mortimer** [reply]
@ericgroom - very much appreciated!!
